### PR TITLE
Also show sampleId if clinical data is missing

### PR DESCRIPTION
**Pull request by inodb**
Fix cbioportal/cbioportal#2831

Before missing clinical data would prevent showing of the sampleId, genomic
overview and VAF plot.

http://www.cbioportal.org/case.do#/patient?sampleId=TCGA-06-0142-01&studyId=gbm_tcga_pub2013

### CODE REVIEW

File: src/pages/patientView/clinicalInformation/getClinicalInformationData.ts [hunk 1]
```diff
@@ -1,99 +0,0 @@
-import URL from 'url';
```
Comment (inodb) [inline src/pages/patientView/clinicalInformation/getClinicalInformationData.ts]: Old code not being used anymore

File: src/pages/patientView/clinicalInformation/lib/convertSamplesData.spec.ts [hunk 2]
```diff
@@ -1,14 +1,14 @@
 import mockClinicalData from "../../../../shared/api/mock/Clinical_data_study_ucec_tcga_pub.json";
 import { ClinicalData } from "../../../../shared/api/generated/CBioPortalAPI";
 import { default as convertSampleData, IConvertedSamplesData } from "./convertSamplesData";
-import { groupByEntityId } from '../getClinicalInformationData';
+import { groupByEntityId } from '../PatientViewPageStore';
```
Comment (inodb) [inline src/pages/patientView/clinicalInformation/lib/convertSamplesData.spec.ts]: we were testing a function that wasn't really used

File: src/pages/patientView/clinicalInformation/PatientViewPageStore.ts [hunk 3]
```diff
@@ -39,12 +39,13 @@ import {
 
 type PageMode = 'patient' | 'sample';
 
-export function groupByEntityId(clinicalDataArray: Array<ClinicalData>) {
+export function groupByEntityId(sampleIds: Array<string>, clinicalDataArray: Array<ClinicalData>) {
```
Comment (onursumer) [inline src/pages/patientView/clinicalInformation/PatientViewPageStore.ts]: Should we move this function into `StoreUtils`?

File: src/pages/patientView/clinicalInformation/PatientViewPageStore.ts [hunk 4]
```diff
@@ -39,12 +39,13 @@ import {
 
 type PageMode = 'patient' | 'sample';
 
-export function groupByEntityId(clinicalDataArray: Array<ClinicalData>) {
+export function groupByEntityId(sampleIds: Array<string>, clinicalDataArray: Array<ClinicalData>) {
+    debugger;
```
Comment (inodb) [inline src/pages/patientView/clinicalInformation/PatientViewPageStore.ts]: oops removing this

**Pull request merged**